ts/Base/MaterialComponent.ts

```diff
@@ -87,12 +96,37 @@ export abstract class MaterialComponent<
     }
   }
 
+  public componentWillUpdate(nextProps: PropType) {
+    if (this.MDComponent && this.mdcNotifyProps) {
+      for (const prop of this.mdcNotifyProps) {
+        if (
+          this.MDComponent.hasOwnProperty(prop) &&
+          this.props[prop] !== nextProps[prop]
+        ) {
+          this.MDComponent[prop as string] = nextProps[prop];
+        }
+      }
+    }
+
+    this.classText = this.buildClassName();
```

Comment:
was this never called till now?

Reply:
Executed a short test currently broken

Reply:
[this](https://github.com/prateekbh/preact-material-components/blob/ecdc66eef95e1662c90bfba56c6330b9c8779ab2/ts/Base/MaterialComponent.ts#L49) should execute on every prop change, no?

Reply:
I can add detection, if a mdcProps was changed

Reply:
it's the same thing but on prop update, I don't exactly know the difference between `render()` and `componentWillUpdate()` (aside that render returns the `VNode`)